crypto/bn/bn_gcd.c

```diff
@@ -609,20 +609,32 @@ int BN_gcd(BIGNUM *r, const BIGNUM *in_a, const BIGNUM *in_b, BN_CTX *ctx)
         || !BN_lshift1(g, in_b)
         || !BN_lshift1(r, in_a))
         goto err;
-
+  
     /* find shared powers of two, i.e. "shifts" >= 1 */
+    pow2_flag = 1;
+    pow2_chunk_index = 0;
+    pow2_shifts = 0;
     for (i = 0; i < r->dmax && i < g->dmax; i++) {
-        mask = ~(r->d[i] | g->d[i]);
-        for (j = 0; j < BN_BITS2; j++) {
-            bit &= mask;
-            shifts += bit;
-            mask >>= 1;
-        }
+        pow2_mask = r->d[i] | g->d[i];
+        pow2_temp = (pow2_flag != 0) & (pow2_mask != 0);
+        pow2_flag &= !pow2_temp;
```

Comment:
I believe there is still some potential for non-const timeness here. I.e. the compiler could transfer this into branches. Could you please use routines from const_time.h appropriately?

Reply:
```cpp
pow2_temp = !constant_time_is_zero(pow2_flag) & !constant_time_is_zero_64(pow2_mask);
```